src/Agent.Listener/DistributedTask.Pipelines/Yaml/Contracts/CheckoutStep.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.TypeConverters;
+
+namespace Microsoft.TeamFoundation.DistributedTask.Orchestration.Server.Pipelines.Yaml.Contracts
+{
+    internal sealed class CheckoutStep : ISimpleStep
+    {
+        public String Name { get; set; }
+
+        internal String Clean { get; set; }
+
+        internal String FetchDepth { get; set; }
+
+        internal String Lfs { get; set; }
+
+        public ISimpleStep Clone()
+        {
+            return new CheckoutStep
+            {
+                Name = Name,
+                Clean = Clean,
+                FetchDepth = FetchDepth,
+                Lfs = Lfs,
+            };
+        }
+
+        internal IList<Variable> GetVariables(IList<ProcessResource> resources)
```

Comment:
This will eventually go away when `checkout` is implemented correctly. My goal with this PR is to set the right pattern to model `checkout` in YAML, even though the underlying implementation what gets generated is hacky right now. Then we can continue to improve the underlying implementation without affecting how it's modeled in YAML.